V0.27  typo tolerance configuration (#1606)
meilisearch/documentation: reference/api/typo_tolerance.md

::: warning
Updating the settings means overwriting the default settings of Meilisearch. You can reset to default values using the `DELETE` routes.
:::

## Get typo tolerance

<RouteHighlighter method="GET" route="/indexes/{index_uid}/settings/typo-tolerance"/>

Get the typo tolerance settings of an index. The index [`uid`](/learn/core_concepts/indexes.md#index-uid) is required.

### Example

<CodeSamples id="get_typo_tolerance_1" />

#### Response: `200 OK`

```json
 {
        "enabled": true,
        "minWordSizeForTypos": {
            "oneTypo": 5,
            "twoTypos": 9
        },
        "disableOnWords": [],
        "disableOnAttributes": []
    }
```

### Returned fields

#### `enabled`

Whether typo tolerance is enabled or not.

#### `minWordSizeForTypos`

Customize the minimum word length for accepting 1 or 2 typos.

| Name       | Description                                  |
|------------|----------------------------------------------|
| `oneTypo`  | The minimum word size for accepting 1 typo  |
| `twoTypos` | The minimum word size for accepting 2 typos |

#### `disableOnWords`

> An array of words on which the typo tolerance feature is disabled.

Reviewer: ```suggestion
An array of words for which the typo tolerance feature is disabled.
```

Reviewer: This sentence sounds a bit wonky, but perhaps I'm just tired 😅

# Typo tolerance

_Child route of the [settings route](/reference/api/settings.md)._

The `typo-tolerance` route allows you to configure the typo tolerance settings for an index.

Typo tolerance settings can also be updated directly through the [global settings route](/reference/api/settings.md#update-settings) along with the other settings.

To learn more about typo tolerance, refer to our [dedicated guide](/learn/configuration/typo_tolerance.md).

::: warning
Updating the settings means overwriting the default settings of Meilisearch. You can reset to default values using the `DELETE` routes.
:::

## Get typo tolerance

<RouteHighlighter method="GET" route="/indexes/{index_uid}/settings/typo-tolerance"/>

Get the typo tolerance settings of an index. The index [`uid`](/learn/core_concepts/indexes.md#index-uid) is required.

### Example

> <CodeSamples id="get_typo_tolerance" />

Reviewer 2: ```suggestion
<CodeSamples id="get_typo_tolerance_1" />
```

#### Response: `200 OK`

```json
 {
        "enabled": true,
        "minWordSizeForTypos": {
            "oneTypo": 5,
            "twoTypos": 9
        },
        "disableOnWords": [],
        "disableOnAttributes": []
    }
```

### Returned fields

#### `enabled`

Whether typo tolerance is enabled or not.

#### `minWordSizeForTypos`

Customize typo tolerance for 1 or 2 typos.

| Name       | Description                                  |
|------------|----------------------------------------------|
| `oneTypo`  | The minimum word size for tolerating 1 typo  |
| `twoTypos` | The minimum word size for tolerating 2 typos |

#### `disableOnWords`

> An array of words with the typo tolerance feature disabled.

Reviewer 3: ```suggestion
An array of words for which typos are not tolerated.
```

#### `disableOnAttributes`

> An array of attributes with the typo tolerance feature disabled.

Reviewer 3: ```suggestion
An array of attributes for which the typo tolerance feature is disabled.
```

## Update typo tolerance

<RouteHighlighter method="POST" route="/indexes/{index_uid}/settings/typo-tolerance"/>

Partially update the typo tolerance settings for an index. The index [`uid`](/learn/core_concepts/indexes.md#index-uid) is required.

### Body

#### `enabled`

**Type:** boolean
**Default value:** `true`

Whether typo tolerance is enabled or not.

#### `minWordSizeForTypos`

**Type:** object

> Customize typo tolerance for 1 or 2 typos.

Reviewer 3: ```suggestion
Customize the minimum word length for which to tolerate 1 or 2 typos.
```

| Name       | Description                                                                   | Type    | Default value |
|------------|-------------------------------------------------------------------------------|---------|---------------|

> | `oneTypo`  | The minimum word size for tolerating 1 typo, must be between `0` and `255`    | integer | `5`           |

Reviewer 3: ```suggestion
| `oneTypo`  | The minimum word size for tolerating 1 typo, must be between `0` and `twoTypos`    | integer | `5`           |
```

> | `twoTypos` | The minimum word size for tolerating 2 typos, must be between `0` and `255`   | integer | `9`           |

Reviewer 3: ```suggestion
| `twoTypos` | The minimum word size for tolerating 2 typos, must be between `oneTypo` and `255`   | integer | `9`           |
```

#### `disableOnWords`

**Type:** array
**Default value:** `[]`

> An array of words with the typo tolerance feature disabled.

Reviewer 3: ```suggestion
An array of words for which typos are not tolerated.
```

#### `disableOnAttributes`

**Type:** array
**Default value:** `[]`

> An array of attributes with the typo tolerance feature disabled.

Reviewer 3: ```suggestion
An array of attributes for which the typo tolerance feature is disabled.
```

#### Example

> <CodeSamples id="update_typo_tolerance" />

Reviewer 2: ```suggestion
<CodeSamples id="update_typo_tolerance_1" />
```

#### Response: `200 OK`

```json

> {
>     "uid": 1,
>     "indexUid": "books",
>     "status": "enqueued",
>     "type": "settingsUpdate",
>     "enqueuedAt": "2022-04-14T20:56:44.991039Z"
>     }
> 

Reviewer 4: ```suggestion
{
    "uid": 1,
    "indexUid": "books",
    "status": "enqueued",
    "type": "settingsUpdate",
    "enqueuedAt": "2022-04-14T20:56:44.991039Z"
}
```

| `oneTypo`  | The minimum word size for tolerating 1 typo  |
| `twoTypos` | The minimum word size for tolerating 2 typos |

#### `disableOnWords`

> An array of words with the typo tolerance feature disabled.

Reviewer 5: ```suggestion
An array of words on which the typo tolerance feature is disabled.
```

Not sure if it's better but it seems clearer, I can't explain it. 😇

#### `disableOnAttributes`

> An array of attributes with the typo tolerance feature disabled.

Reviewer 5: ```suggestion
An array of attributes on which the typo tolerance feature is disabled.
```

> An array of words with the typo tolerance feature disabled.

Reviewer 5: ```suggestion
An array of words on which the typo tolerance feature is disabled.
```

#### `disableOnAttributes`

**Type:** array
**Default value:** `[]`

> An array of attributes with the typo tolerance feature disabled.

Reviewer 5: ```suggestion
An array of attributes on which the typo tolerance feature is disabled.
```
